First run format, then lint

When configuring a package, the code runs `tox -p auto`, which is what Zope does too. Currently for our setup this means `format` and `lint`. This is confusing, because `lint` will complain about code that is being changed in parallel by `format`. For example see this output:

```
$ bin/python config-package.py --no-push ../../6.0/src/plone.batching
plone.batching is already configured for this config type, updating.
format: FAIL ✖ in 0.24 seconds
format: commands[0]> sh -c 'find plone -name "*.py" | xargs pyupgrade --py38-plus *.py'
format: exit 1 (0.16 seconds) /Users/maurits/community/plone-coredev/6.0/src/plone.batching> sh -c 'find plone -name "*.py" | xargs pyupgrade --py38-plus *.py' pid=46845
lint: commands[0]> sh -c 'find plone -name "*.py" | xargs flake8 *.py'
conf.py:13:1: F401 'sys' imported but unused
...
plone/batching/batch.py:276:44: E127 continuation line over-indented for visual indent
plone/batching/batch.py:286:1: E305 expected 2 blank lines after class or function definition, found 1
plone/batching/browser.py:63:1: E303 too many blank lines (3)
lint: exit 1 (0.62 seconds) /Users/maurits/community/plone-coredev/6.0/src/plone.batching> sh -c 'find plone -name "*.py" | xargs flake8 *.py' pid=46846
  format: FAIL code 1 (0.24=setup[0.08]+cmd[0.16] seconds)
  lint: FAIL code 1 (0.69=setup[0.07]+cmd[0.62] seconds)
  evaluation failed :( (0.77 seconds)
Rewriting conf.py
Rewriting setup.py
Rewriting plone/__init__.py
Rewriting plone/batching/interfaces.py
Rewriting plone/batching/batch.py
Rewriting plone/batching/__init__.py
Rewriting plone/batching/browser.py
Rewriting plone/batching/utils.py
Rewriting plone/batching/tests.py
ABORTING: Please fix the errors shown above.
Proceed anyway (y/N)?
```

When I then continue anyway, and run lint by hand, it no longer complains. Okay, wait, it still complains about a few things that need manual fixing. But the point is valid: first run `format`, then `lint`, either separately or in one command:

```
tox -e format,lint
```

--- a/config/config-package.py
+++ b/config/config-package.py
@@ -207,6 +207,24 @@ class PackageConfiguration:
 
         return destination.relative_to(self.path)
 
+    def remove_old_files(self):
+        filenames = ('bootstrap.py', '.travis.yml')
+        with change_dir(self.path):
+            for filename in filenames:
+                if pathlib.Path(filename).exists():
+                    call('git', 'rm', 'filename')
+
+    def remove_toml_empty_sections(self):
+        meta_cfg = {k: v for k, v in self.meta_cfg.items() if v}
+        with change_dir(self.path):
+            with open('.meta.toml', 'w') as meta_f:
+                meta_f.write(META_HINT.format(config_type=self.config_type))
+                meta_f.write('\n')
+                toml.dump(
+                    meta_cfg, meta_f,
+                    TomlArraySeparatorEncoderWithNewline(
+                        separator=',\n   ', indent_first_line=True))
+
     def configure(self):
         self._add_project_to_config_type_list()
 
@@ -219,21 +237,10 @@ class PackageConfiguration:
             self.tox(),
         ]
 
-        with change_dir(self.path) as cwd:
-            if pathlib.Path('bootstrap.py').exists():
-                call('git', 'rm', 'bootstrap.py')
-            if pathlib.Path('.travis.yml').exists():
-                call('git', 'rm', '.travis.yml')
-            # Remove empty sections:
-            meta_cfg = {k: v for k, v in self.meta_cfg.items() if v}
-            with open('.meta.toml', 'w') as meta_f:
-                meta_f.write(META_HINT.format(config_type=self.config_type))
-                meta_f.write('\n')
-                toml.dump(
-                    meta_cfg, meta_f,
-                    TomlArraySeparatorEncoderWithNewline(
-                        separator=',\n   ', indent_first_line=True))
+        self.remove_old_files()
+        self.remove_toml_empty_sections()
 
+        with change_dir(self.path) as cwd:
             tox_path = shutil.which('tox') or (
                 pathlib.Path(cwd) / 'bin' / 'tox')
             call(tox_path, '-p', 'auto')

--- a/config/config-package.py
+++ b/config/config-package.py
@@ -230,6 +230,29 @@ class PackageConfiguration:
             tox_path = shutil.which('tox') or (pathlib.Path(cwd) / 'bin' / 'tox')
             call(tox_path, '-e', 'format,lint')
 
+    @property
+    def _commit_msg(self):
+        return self.args.commit_msg or 'Configuring with plone/meta'
+
+    def commit_and_push(self, filenames):
+        if not self.args.commit:
+            return
+
+        with change_dir(self.path):
+            call('git', 'add', *filenames)
+            call('git', 'commit', '-m', self._commit_msg)
+            if self.args.push:
+                call('git', 'push', '--set-upstream', 'origin', self.branch_name)
+
+    @staticmethod
+    def final_help_tips(updating):
+        print()
+        print('If everything went fine up to here:')
+        if updating:
+            print('Updated the previously created PR.')
+        else:
+            print('Create a PR, using the URL shown above.')
+
     def configure(self):
         self._add_project_to_config_type_list()
 
@@ -249,22 +272,8 @@ class PackageConfiguration:
         with change_dir(self.path):
             updating = git_branch(self.branch_name)
 
-            if self.args.commit:
-                call('git', 'add', *files_changed)
-                if self.args.commit_msg:
-                    commit_msg = self.args.commit_msg
-                else:
-                    commit_msg = f'Configuring with plone/meta'
-                call('git', 'commit', '-m', commit_msg)
-                if self.args.push:
-                    call('git', 'push', '--set-upstream',
-                         'origin', self.branch_name)
-            print()
-            print('If everything went fine up to here:')
-            if updating:
-                print('Updated the previously created PR.')
-            else:
-                print('Create a PR, using the URL shown above.')
+        self.commit_and_push(files_changed)
+        self.final_help_tips(updating)
 
 
 def main():

--- a/config/config-package.py
+++ b/config/config-package.py
@@ -225,6 +225,11 @@ class PackageConfiguration:
                     TomlArraySeparatorEncoderWithNewline(
                         separator=',\n   ', indent_first_line=True))
 
+    def run_tox(self):
+        with change_dir(self.path) as cwd:
+            tox_path = shutil.which('tox') or (pathlib.Path(cwd) / 'bin' / 'tox')
+            call(tox_path, '-e', 'format,lint')
+
     def configure(self):
         self._add_project_to_config_type_list()
 
@@ -239,12 +244,9 @@ class PackageConfiguration:
 
         self.remove_old_files()
         self.remove_toml_empty_sections()
+        self.run_tox()
 
-        with change_dir(self.path) as cwd:
-            tox_path = shutil.which('tox') or (
-                pathlib.Path(cwd) / 'bin' / 'tox')
-            call(tox_path, '-p', 'auto')
-
+        with change_dir(self.path):
             updating = git_branch(self.branch_name)
 
             if self.args.commit:

